### PR TITLE
Week1

### PR DESCRIPTION
**Pull request by oishikg**
See file `hw1/Mylib.hs` for solutions; questions are in the comments but the pdf itself may be found in 
`hw1`

### CODE REVIEW

File: week1/hw1/MyLib.hs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+-}
+
+{- Reverse the given list; this way, we start at the last digit; then pattern match
+and at each recursive call, double the second digit; if there is no second digit
+then return the list as it is -} 
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOtherHelper :: [Integer] -> [Integer]
+
+doubleEveryOtherHelper [] = []
+doubleEveryOtherHelper (x : []) = (x : []) -- is there any 'as' construct?
```
Comment (arbus) [inline week1/hw1/MyLib.hs]: Yes the equivilant of as is @

```suggestion
doubleEveryOtherHelper (x : [])@orig = orig
```

File: week1/hw1/MyLib.hs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+and at each recursive call, double the second digit; if there is no second digit
+then return the list as it is -} 
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOtherHelper :: [Integer] -> [Integer]
+
+doubleEveryOtherHelper [] = []
+doubleEveryOtherHelper (x : []) = (x : []) -- is there any 'as' construct? 
+doubleEveryOtherHelper (x : x' : xs') =  x : (2 * x') : (doubleEveryOtherHelper xs')
+
+doubleEveryOther xs = revList (doubleEveryOtherHelper (revList xs))
```
Comment (arbus) [inline week1/hw1/MyLib.hs]: When feeding the value of one function to the other, try touse function composition as this leads to much cleaner code. 

```suggestion
doubleEveryOther = revList . doubleEveryOtherHelper . revList
```

File: week1/hw1/MyLib.hs~ [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module MyLib (someFunc) where
```
Comment (arbus) [inline week1/hw1/MyLib.hs~]: remove

File: week1/hw1/TAGS [hunk 4]
```diff
@@ -0,0 +1,55 @@
+
```
Comment (arbus) [inline week1/hw1/TAGS]: remove

File: week1/hw1/hw1.cabal~ [hunk 5]
```diff
@@ -0,0 +1,32 @@
+cabal-version:       2.4
```
Comment (arbus) [inline week1/hw1/hw1.cabal~]: remove

File: week1/lecture1/TAGS [hunk 6]
```diff
@@ -0,0 +1,111 @@
+
```
Comment (arbus) [inline week1/lecture1/TAGS]: remove

File: week1/hw1/MyLib.hs [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+{- At each recursive call, take the remainer of the number when divided with 10; this
+gives the last digit of that number; append a list with the digit to the digits of the
+quotient -}
+toDigitsHelper :: Integer -> [Integer] -> [Integer]
+toDigitsHelper n acc
+  | n <= 0 = acc
+  | otherwise = toDigitsHelper (n `div` 10) ((n `mod` 10) : acc)
+
+toDigits :: Integer -> [Integer]
+toDigits n = toDigitsHelper n []
```
Comment (JolandaNava) [inline week1/hw1/MyLib.hs]: Just for your amusement: if you define `toDigitsHelper` as ` [Integer] -> Integer -> [Integer]` (e.g. swap the order of the arguments), you can then write 
```
toDigits :: Integer -> [Integer]
toDigits = toDigitsHelper []
```

File: week1/hw1/MyLib.hs [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+of digits and sum them up -}
+
+sumDigits :: [Integer] -> Integer
+sumDigitsHelper :: [Integer] -> Integer
+
+sumDigitsHelper [] = 0
+sumDigitsHelper (x : xs) = x + sumDigitsHelper xs 
+
+sumDigits [] = 0
+sumDigits (x : xs) = sumDigitsHelper (toDigits x) + sumDigits xs
```
Comment (JolandaNava) [inline week1/hw1/MyLib.hs]: You can have a look at [concatMap](https://hackage.haskell.org/package/base-4.12.0.0/docs/GHC-List.html#v:concatMap) if you want to try and write this as a composition of functions. ([sum](https://hackage.haskell.org/package/base-4.12.0.0/docs/GHC-List.html#v:sum) might also come in handy)

**Pull request merged**